### PR TITLE
Fix false positive for rule `BA2024.EnableSpectreMitigations` incorrectly flags compilation units using debug runtime (which are not Spectre-mitigated by design).

### PR DESCRIPTION
**Pull request by shaopeng-gh**
BinSkim’s BA2024 rule warns on debug VC runtime libraries, e.g.:

```
'xxx.dll' was compiled with one or more modules that do not enable code generation mitigations for speculative execution side-channel attack (Spectre) vulnerabilities. Spectre attacks can compromise hardware-based isolation, allowing non-privileged users to retrieve potentially sensitive data from the CPU cache. To resolve the issue, provide the /Qspectre switch on the compiler command-line (or /d2guardspecload in cases where your compiler supports this switch and it is not possible to update to a toolset that supports /Qspectre). This warning should be addressed for code that operates on data that crosses a trust boundary and that can affect execution, such as parsing untrusted file inputs or processing query strings of a web request.
The following modules were compiled with a toolset that supports /Qspectre but the switch was not enabled on the command-line:
MSVCRTD.lib,cxx,19.33.31616.1 (…)
MSVCRTD.lib,c,19.33.31616.1 (…)
vcruntimed.lib,cxx,19.33.31616.1 (…)
```

Since those debug libraries are not Spectre-mitigated by design, I think we should either suppress warnings for those libraries globally.  For the record, there is BinSkim issue on better handling this case:
https://github.com/microsoft/binskim/issues/718

### CODE REVIEW

File: src/Test.FunctionalTests.BinSkim.Driver/BaselineTestData/Expected/clangcl.pe.cpp.codeview.exe.sarif [hunk 1]
```diff
@@ -381,11 +381,12 @@
         {
           "ruleId": "BA2024",
           "ruleIndex": 16,
+          "kind": "pass",
```
Comment (shaopeng-gh) [inline src/Test.FunctionalTests.BinSkim.Driver/BaselineTestData/Expected/clangcl.pe.cpp.codeview.exe.sarif:384]: >pass [](http://example.com/codeflow?start=19&length=4)

In the baseline changes,
all built with debug (including this one with -Mtd)
was failing now will pass.

File: src/BinSkim.Rules/PERules/BA2024.EnableSpectreMitigations.cs [hunk 2]
```diff
@@ -145,6 +145,9 @@ public override void AnalyzePortableExecutableAndPdb(BinaryAnalyzerContext conte
                 Symbol om = omView.Value;
                 ObjectModuleDetails omDetails = om.GetObjectModuleDetails();
 
+                // Debug libraries are not Spectre-mitigated by design.
```
Comment (schlaman-ms) [inline src/BinSkim.Rules/PERules/BA2024.EnableSpectreMitigations.cs:148]: Small nit for fast readability and consistency:
I'd rather see:

if (.....)
{
    continue;
}

Comment (shaopeng-gh) [reply src/BinSkim.Rules/PERules/BA2024.EnableSpectreMitigations.cs:148]: fixed.

**Review by schlaman-ms — APPROVED** [review]
:shipit:

**Review by suvamM — APPROVED** [review]
:shipit:

**Pull request merged**